lib/hanami/routing/route.rb

```diff
@@ -50,7 +50,8 @@ def generate(resolver, options = {}, &endpoint)
       # @since 0.2.0
       # @api private
       def nested_router
-        dest.routes if dest.respond_to?(:routes)
+        dest.routes if dest.respond_to?(:routes) &&
+          dest.routes.is_a?(Hanami::Router)
```

Comment:
@cyphactor What if instead of checking for type, we still check for behavior? In a language like Ruby, this is always an advantage for openness.

For instance:

```ruby
def nested_router
  if dest.respond_to?(:routes) && dest.routes.respond_to?(:inspector)
    dest.routes
  end
end
```

Do you think it can still work?